### ISSUE TITLE
Modifier l'espacement entre deux lignes (text_show)

### ISSUE THREAD

**Issue by YendaHusk** [RMEx/RME#470]
### Configuration (lorsque le bug a été rencontré)

- **Version de RME** : `2.0.0`
- **Commande** :  `text_show`
- *Autres scripts (s'il y en a*) : `Orms`

---

### Description du bug

Hello ! Je travail sur un système de dialogue custom via la commande text_show et text_progressive.
J'aimerais pouvoir stocké mes dialogues en une seul ligne dans un fichier texte. Ça fonctionne sans aucun souci de ce côté.
Par contre, pour faire des dialogues plus long j'aimerais utilisé le retour à la ligne. Ce que je fait via la commande  `\n`
Le problème que je rencontre, c'est l'espacement entre les deux lignes qui est beaucoup trop grand.
Du coup je me demander si il été possible de configuré ça, ou alors de le forcer d'une manière ou d'une autre ?
![image](https://www.mediafire.com/convkey/8dd3/7hce7iln102b4cyzg.jpg?size_id=5)

**Comment by BilouMaster** [comment]
Ca peut être une forme de conflit entre les deux scripts (ORMS / RME, les deux manipulent la valeur d'espacement du texte) puisque l'espacement attendu devrait être égal à celui d'un texte normal (Commande événement > Message) avec ou sans ORMS.

Je suis un peu occupé là... (comme toujours) alors si tu veux tordre un peu (provisoirement) les paramètres de ORMS pour comparer la différence entre text_show et la commande événement habituelle, ça pourrait déjà t'aider à mieux préciser les paramètres qui entrent en jeux.

Sinon un petit patch et on n'en parle plus. ;)

j.🌸

**Comment by YendaHusk** [comment]
"Malheureusement" le problème ne viens pas d'Orms, même sans, le décalage est toujours présent.
J'ai essayer de jouer un peu avec les valeurs de Orms, notamment `FONT_HEIGHT` et `LINE_HEIGHT`
La première ne donne évidemment rien de bon et la deuxième change effectivement les espacements entre les lignes, mais pour les menus uniquement, ça n'affecte pas les retours à la ligne. =/
![image](https://www.mediafire.com/convkey/dd92/fbilva5rpavh7dozg.jpg)

**Comment by xvw** [comment]
C'est un bug. L'espacement entre les textes n'est pas calculé de la même manière que les messages.

**Comment by YendaHusk** [comment]
Il y aurais un moyen de remédier à ça temporairement avec un patch ou autres ?

**Comment by BilouMaster** [comment]
Bien sûr, mais @xvw semble occupé. J'allais chercher la ligne où était calculé ce fameux espacement pour comprendre sur quoi xvw s'était basé (si ce n'est pas celui par défaut, j'imagine un espacement configuré dans le profil de la font).

Si j'y pense, je verrais pour bricoler un patch demain, sinon si tu trouve toi-même, n'hésite pas à patcher toi-même et partager ici la manière dont tu as patché... C'est bon d'avoir une certaine vision de comment fonctionnent les scripts que tu ajoutes à ton projet et de savoir les patcher toi-même si besoin. (Et encore plus cool de partager tes patchs qui pourraient inspirer des corrections, ajouts ou modifications pour la prochaine MàJ si le projet est encore maintenu.)

**Comment by YendaHusk** [comment]
Du coup j'ai chercher dans les 6800 lignes de codes d'rme, j'ai pas réussi à trouver (ou à comprendre) la ligne qui gère ça. Si vous savez comment la trouver je veux bien essayez par moi-même de bidouiller ça. Même si je doute d'en avoir les compétences.. :D

**Comment by pytre** [comment]
6800 ? RME fait ~17400 lignes (sans la documentation) ! 
La méthode qui gère les textes se trouve ici : https://github.com/RMEx/RME/blob/master/src/EvEx.rb#L3455

Je pense que ici : https://github.com/RMEx/RME/blob/master/src/EvEx.rb#L3467 : 

```ruby
width, height = widths.max + font.size, heights.max + font.size
```
le dernier +font.size semble être de trop. 

Quoi qu'il en soit, je pense qu'il faudrait réécrire "un peu" le support de textes et sortir l'attribut `size` du profil.

**Comment by YendaHusk** [comment]
Ah oui effectivement, j'avais pas fait attention qu'une fois rendu à 9999, le compteur reprenait à 0.
Du coup merci beaucoup ! Le `+ font.size` était effectivement en trop.
Je l'ai seulement ajouter à la ligne du dessous pour faire en sorte que si le texte fait deux ligne, il ne viennes pas se superposé sur le personnage.
```
    width, height = widths.max + font.size, heights.max
    total_height = height * lines.length + font.size
```
Merci de votre aide à tous ! :)
Et vive RME ! :D

**Comment by xvw** [comment]
Merci @pytre ! 
Effectivement, je corrigerai ça dans la version 2.2.0. 
Je suis aussi d'accord sur le fait de retirer la taille des profiles (et d'ajouter un interligne).
